app/javascript/components/verifiable.js

```diff
@@ -27,12 +31,16 @@ export default template({
     },
     onChangeReferenceURL: function() {
       this.$emit('reference-url-change', this.userReferenceURL)
+      localStorage.setItem(this.localStorageKey, this.userReferenceURL)
       this.editing = false
     }
   },
   computed: {
     referenceURL: function() {
       return this.statement.reference_url || this.page.reference_url
+    },
+    localStorageKey: function() {
+      return this.page.position_held_item + '-reference_url'
```

Comment:
I think I'd make this the page title, since there can be multiple pages for a position_held_item, and you might well want to have different reference URLs that you'd use on each. (This would also be consistent with what we do for Wikipedia selection: https://github.com/mysociety/verification-pages/blob/b5f0d23ed1e17ccd0ba1806aad08794af0bd0e43/app/javascript/components/reconcilable.js#L143 )